component/prometheus/exporter/exporter.go

```diff
@@ -29,36 +29,38 @@ type Component struct {
 
 	reload chan struct{}
 
-	creator Creator
+	creator         Creator
+	multiTargetFunc func(discovery.Target, component.Arguments) []discovery.Target
+	baseTarget      discovery.Target
 
 	exporter       integrations.Integration
 	metricsHandler http.Handler
 }
 
-func New(creator Creator, name string) func(component.Options, component.Arguments) (component.Component, error) {
+func New(creator Creator, name string, multiTargetFunc func(discovery.Target, component.Arguments) []discovery.Target) func(component.Options, component.Arguments) (component.Component, error) {
```

Comment:
wondering if it would make sense to add a separate constructor for this and refactor the code to avoid duplication. The idea is to avoid increasing the complexity of the base case for something that is quite niche and specific to this and maybe very few other exporters

Reply:
something like 

```golang
func NewMultiTarget(creator Creator, name string, multiTargetFunc func(discovery.Target, component.Arguments) []discovery.Target)
```

while leaving the other function untouched